pkg/controller/v1beta1/inferenceservice/components/predictor/predictor.go

```diff
@@ -176,6 +180,29 @@ func (p *Predictor) CreatePredictorService(isvc *v1beta1.InferenceService) (*kns
 		},
 	}
 
+	//If InferenceService's storageUri is empty, create multi-model service configMap and
+	//mount it into this predictor's knative service
+	storageUri := isvc.Spec.Predictor.GetStorageUri()
+	if storageUri != nil && len(*storageUri) > 0 {
```

Comment:
Also -- would it be cleaner to write a configmap in all cases? If the inferenceservice has a storageUri, it can just write the configmap itself. That would make your storage downloader implementation easier, as it would only have one code path to contend with. It would simply need to understand the configmap.

Reply:
Hi, I actually thought about it before! If we do it this way then every InferenceService is a multi-model server and the InferenceService with a storageUri is just a special case. However, we want to only mount the puller sidecar for InferenceServices without storageUri to save resource overhead. If we choose to do that, I'd. prefer to use the storageUri as the flag to differentiate between MMS InferenceService and single model InferenceService. @yuzisun WDYT?

Reply:
> Validation logic should prevent the length of storageUri being 0, so you could simplify this to just checking existence.
> 
> if isvc.Spec.Predictor.GetStorageUri() != nil {
> }

Agree! But checking nil also doesn't hurt here :)

Reply:
> Hi, I actually thought about it before! If we do it this way then every InferenceService is a multi-model server and the InferenceService with a storageUri is just a special case. However, we want to only mount the puller sidecar for InferenceServices without storageUri to save resource overhead. If we choose to do that, I'd. prefer to use the storageUri as the flag to differentiate between MMS InferenceService and single model InferenceService. @yuzisun WDYT?

We need a sidecar either way. The other way is an init container for model downloading. Sure, the new KFServing Agent sidecar runs forever rather than as an init container, but I don't see this as a big performance issue. In fact, you could run the KFServing agent as an init container in the single model server case. 

Either way, I think we should write a configmap in all cases to simplify this architecture.